api_app/connectors_manager/api.py

```diff
@@ -0,0 +1,20 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+from rest_framework import generics
+from rest_framework.response import Response
+
+from api_app.connectors_manager.serializers import ConnectorConfigSerializer
+from api_app.connectors_manager import helpers
+
+
+class ConnectorListAPI(generics.ListAPIView):
```

Comment:
Can we do this using `Viewsets`? If so, any particular reason of choosing one over the other for our case?

Reply:
We use viewsets when we have corresponding models using which we can define a `queryset`. But since we don't have model for Connector, we use a generic view.

- In a viewset, you define methods like `create`, `list`, `destroy` which corresponds to the HTTP methods `POST`, `GET`, `DELETE` and performs operations on the defined queryset and model. (mostly you don't have to define these methods yourself because DRF takes care of it)
- In a view class, you can define methods like `get`, `post` etc. which correspond to the HTTP methods. So this is useful when you wish to define a custom view that does not necessarily depend on a model/queryset.

For further clarification, please refer to the DRF docs, they are pretty well written.